tests/1_api.js

```diff
@@ -48,490 +48,563 @@ describe('Resources', function() {
 
 	afterEach(function() {
 		xhr.restore();
+		clock.restore();
 		server.createScript.restore();
 	});
 
-	describe('/v1/open', function() {
-		it('should pass in app_id and browser_fingerprint_id', function(done) {
-			var assert = testUtils.plan(5, done);
-			server.request(resources.open, testUtils.params({ "is_referrable": 1 }), storage(), assert.done);
-			assert.equal(requests.length, 1, 'Request made');
-			assert.equal(requests[0].url, 'https://api.branch.io/v1/open', 'Endpoint correct');
-			assert.equal(requests[0].method, 'POST', 'Method correct');
-			assert.equal(requests[0].requestBody, "app_id=" + app_id + "&identity_id=" + identity_id + "&is_referrable=1&browser_fingerprint_id=" + browser_fingerprint_id, 'Data correct');
-			requests[0].respond(200, { "Content-Type": "application/json" }, '{ "session_id": 123 }');
+	describe('XHR Request', function() {
+		beforeEach(function() {
+			sinon.stub(goog.json, "parse", function(data) {
+				return data;
+			});
 		});
 
-		it('should pass as a jsonp request', function(done) {
-			storage()['setItem']('use_jsonp', true);
-
-			var assert = testUtils.plan(3, done);
+		afterEach(function() {
+			goog.json.parse.restore();
+		});
 
-			var completeParams = testUtils.params({ "is_referrable": 1 });
-			server.request(resources.open, completeParams, storage(), assert.done);
+		it('should instantiate an XHR', function(done) {
+			var assert = testUtils.plan(4, done);
+			var spyCallback = sinon.spy();
+			server.XHRRequest(resources.profile, testUtils.params({ "identity": "test_id" }), 'POST', storage(), spyCallback);
 			assert.equal(requests.length, 1, 'Request made');
-
-			var encodedData = encodeURIComponent(utils.base64encode(goog.json.serialize(completeParams)));
-			assert.equal(requests[0].src, 'https://api.branch.io/v1/open?&data=' + encodedData + '&callback=branch_callback__' + (server._jsonp_callback_index - 1), 'Endpoint correct');
-
-			requests[0].callback();
+			assert.equal(requests[0].timeout, 5000, 'timeout set to 5s');
+			assert.equal(requests[0].requestHeaders['Content-Type'], 'application/x-www-form-urlencoded;charset=utf-8');
+			requests[0].ontimeout();
```

Comment:
instead of calling this directly, i think you can just move the clock forward >5 seconds and it should work right?